src/shared/sleep-config.c

```diff
@@ -271,6 +356,48 @@ int put_battery_discharge_rate(int estimated_battery_discharge_rate) {
         return 0;
 }
 
+/* Estimate battery discharge rate using stored previous and current capacity over timestamp difference */
+int estimate_battery_discharge_rate_per_hour(
+                                        Hashmap *last_capacity,
+                                        Hashmap *current_capacity,
+                                        usec_t before_timestamp,
+                                        usec_t after_timestamp) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        uint64_t system_hash_id;
+        sd_device *dev;
+        int battery_last_capacity, battery_current_capacity, battery_discharge_rate, r;
+
+        r = get_sd_device_battery(&e);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get available battery: %m");
+
+        FOREACH_DEVICE(e, dev) {
+                const char *x = NULL;
+
+                r = sd_device_get_property_value(dev, "POWER_SUPPLY_NAME", &x);
+                if (r == -ENOENT)
+                        return log_device_debug_errno(dev, r, "battery capacity is unavailable. Assuming no battery exists: %m");
```

Comment:
this should log and continue

Reply:
same as above. this should be skipped and we do not need to seperately handle -ENOENT.